### PR TITLE
Payment status

### PR DESCRIPTION
**Pull request by JakubSQ**
Opis:
As an admin, I want to be able to manually update the order's payment status so it can be shipped and completed.

Acceptance criteria:
- New orders have a "pending" payment status
- Admin can change an order's payment status from "pending" to "failed" or "completed" using a button on the order's page
- There is a system & request spec

### CODE REVIEW

File: app/controllers/admin/orders_controller.rb [hunk 1]
```diff
@@ -8,4 +8,18 @@ def index
   def show
     @order = Order.find(params[:id])
   end
+
+  def order_status
+    @order = Order.find(params[:id])
```
Comment (kamilpuchala) [inline app/controllers/admin/orders_controller.rb:13]: @order = Order.find(params[:id]) can you extract this to the separate method?

File: app/views/admin/orders/show.html.erb [hunk 2]
```diff
@@ -1,10 +1,32 @@
-<%= render partial: 'flash_messages' %>
 
 <div class="card">
   <div class="card-header">
     Order id: <%= @order.id %> set by: <%= @order.user_email %> (user id: <%= @order.user_id %>) at <%= @order.created_at %> with current state: <%= @order.state %>
   </div>
   <div class="card-body">
+  <p class="card-text">Order Status: <%= @order.state %></p>
+      <% if @order.aasm.permitted_transitions.any? %>
```
Comment (kamilpuchala) [inline app/views/admin/orders/show.html.erb:8]: Looks like a lot of logic here. 
Could you refactor it? decorator? and simplify the if statement?

File: spec/requests/order_page_request_spec.rb [hunk 3]
```diff
@@ -6,7 +6,8 @@
   describe 'GET orders#show' do
     let(:admin) { create(:admin) }
     let(:user) { create(:user) }
-    let!(:order) { Order.create(user_id: user.id) }
+    let!(:payment) { Payment.create }
```
Comment (kamilpuchala) [inline spec/requests/order_page_request_spec.rb:9]: Can you move it to a factory?same for order

File: spec/requests/payment_status_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'AdminOrderPaymentStatus', type: :request do
+  describe 'GET orders#show' do
+    let(:admin) { create(:admin) }
+    let(:user) { create(:user) }
+    let!(:payment) { Payment.create }
```
Comment (kamilpuchala) [inline spec/requests/payment_status_request_spec.rb:9]: As in previous spec

File: spec/system/order_page_spec.rb [hunk 5]
```diff
@@ -7,8 +7,9 @@
   let(:user) { create(:user) }
   let!(:cart) { Cart.create }
   let!(:product) { create(:product) }
+  let!(:payment) { Payment.create }
```
Comment (kamilpuchala) [inline spec/system/order_page_spec.rb:10]: same here, and in other specs :)

**Review by kamilpuchala — COMMENTED** [review]
Looks ok, please check my comments

File: app/controllers/admin/orders_controller.rb (hunk 1 above)
Comment (JakubSQ) [reply app/controllers/admin/orders_controller.rb:13]: I don' understand. Should I create before_action method? As far as I remember you didn't recommend this kind of approach.

File: app/views/admin/orders/show.html.erb (hunk 2 above)
Comment (JakubSQ) [reply app/views/admin/orders/show.html.erb:8]: This logic is implemented according gem docs.
# List all the permitted transitions(event and state pairs) from initial state
job.aasm.permitted_transitions
#=> [{ :event => :run, :state => :running }]

**Review by kamilpuchala — APPROVED** [review]

**Pull request merged**